docs/cheat-sheet/cheat-sheet.livecodescript

```diff
@@ -0,0 +1,739 @@
+   put it into tGroup
+   
+   create field "title" in tGroup
+   put it into tTitle
+   set the width of tTitle to pWidth
+   set the text of tTitle to pContent["title"]
+   set the textSize of tTitle to kSectionTitleTextSize
+   set the textStyle of tTitle to "underline"
+   set the textAlign of tTitle to "center"
+   
+   create field "text" in tGroup
+   put it into tText
+   set the width of tText to pWidth
+   set the text of tText to pContent["text"]
+   set the textSize of tText to kTextSize
+   set the textAlign of tText to "left"
+   
+   if pContent["text"] is empty then
+      set the height of tText to 0
+      set the bottom of tText to the bottom of tTitle
+      hide tText
+   else
+      set the height of tText to the formattedHeight of tText
+      set the top of tText to the bottom of tTitle + kTitlePadding
+   end if
+   
+   create field "code" in tGroup
+   put it into tCode
+   set the width of tCode to pWidth
+   set the text of tCode to pContent[LanguageToKey("livecode")]
+   set the textSize of tCode to kCodeTextSize
+   
+   _internal script colorize char 1 to the number \ 
+         of chars in pContent[LanguageToKey("livecode")] of tCode
+   set the height of tCode to the formattedHeight of tCode
+   set the loc of tCode to the loc of tTitle
+   set the top of tCode to the bottom of tText + kPadding
+end BuildGroupForLiveCode
+
+command PositionRowsInColumn @xCol, pColWidth, pTitleBottom, @xContentBottom, pContent, @xToFit
+   local tLastBottom, tLeft
+   put xCol*kMargin + (xCol - 1)*pColWidth into tLeft
+   
+   local tCurGroup, tLines, tPositioned
+   put false into tPositioned
+   put xToFit into tLines
+   repeat for each line tLine in tLines
+      put pContent[tLine]["title"] into tCurGroup
+      if there is not a group tCurGroup then
+         delete line (lineoffset(tLine, xToFit)) of xToFit
+         next repeat
+      end if
+      
+      local tRowHeight
+      put RowHeight(tCurGroup) into tRowHeight
+      if tRowHeight + xContentBottom > the height of me then
+         next repeat
+      end if
+      
+      set the top of group tCurGroup to xContentBottom
+      set the left of group tCurGroup to tLeft
+      add tRowHeight to xContentBottom
+      delete line (lineoffset(tLine, xToFit)) of xToFit
+      put true into tPositioned
+   end repeat
+   
+   if tPositioned then
+      exit PositionRowsInColumn
+   end if
+   
+   // If we get here we need a new column
+   add 1 to xCol
+   put pTitleBottom into xContentBottom
+end PositionRowsInColumn
+
+function CheatSheetArray
+   local tPath
+   put the effective filename of me into tPath
+   set the itemdelimiter to slash
+   put "cheat-sheet.yml" into item -1 of tPath
+   
+   local tYAML
+   put textDecode(url("binfile:" & tPath), "utf8") into tYAML
+   
+   YAMLToArray tYAML
+   if the result is not empty then
+      return the result
+   end if
+   return it
+end CheatSheetArray
+
+## YAMLToArray by Monte Goulding
```

Comment:
Heh, yes - no worries for now but it would be good. I had to jump through a couple of small hoops to get the yaml to comply with the subset you support here!